js/app.js

```diff
@@ -0,0 +1,548 @@
+                </label>
+            </div>
+        );
+    }
+});
+
+var Setting = React.createClass({
+    handleChange: function(event) {
+        var objectKey = this.props.objectKey,
+            type = event.target.dataset.type,
+            data = this.props.data;
+
+        this.props.onChange(objectKey, type, data);
+    },
+    render: function() {
+        return (
+            <div className="checkbox">
+                <label>
+                    <input
+                        type="checkbox"
+                        className="setting"
+                        name="setting"
+                        data-type="settings"
+                        checked={this.props.checked}
+                        onChange={this.handleChange}
+                        disabled={this.props.disabled} />
+                            {this.props.longName}
+                </label>
+            </div>
+        );
+    }
+});
+
+var OutputType = React.createClass({
+    handleChange: function(event) {
+        var objectKey = this.props.objectKey,
+            type = event.target.dataset.type,
+            data = this.props.data;
+
+        this.props.onChange(objectKey, type, data);
+    },
+    render: function() {
+        return (
+            <div className="radio">
+                <label>
+                    <input
+                        type="radio"
+                        className="outputType"
+                        name="outputType"
+                        data-type="outputTypes"
+                        checked={this.props.checked}
+                        onChange={this.handleChange} />
+                            {this.props.name}
+                </label>
+            </div>
+        );
+    }
+});
+
+var generateOutput = function(selectedOutputType, columns, settings, data) {
+    var output = "";
+    var selectedColumns = [];
+    var columnsDefinition = "";
+    var countries = "";
+
+    for (var key in columns) {
+        if (columns.hasOwnProperty(key) && columns[key].checked) {
+            selectedColumns.push(columns[key]);
+        }
+    }
+
+    switch (selectedOutputType) {
+        case OUTPUT_MYSQL:
+            if (settings.languagelookup.checked) {
+                var lookupCountries = "";
+                var lookupColumnsDefinition = "";
+
+                output =
+                    "CREATE TABLE IF NOT EXISTS `countries` (\n" +
+                    "    `id` int(5) NOT NULL AUTO_INCREMENT,\n" +
+                    "{0}" +
+                    "    PRIMARY KEY (`id`)\n" +
+                    ") ENGINE=MyISAM DEFAULT CHARSET=utf8 AUTO_INCREMENT=0;\n\n" +
+                    "" +
+                    "CREATE TABLE IF NOT EXISTS `country_lang_lnk` (\n" +
+                    "{1}" +
+                    "    PRIMARY KEY (`countryCode`,`languages`)\n" +
+                    ") ENGINE=MyISAM DEFAULT CHARSET=utf8;\n\n" +
+                    "{2}\n" +
+                    "{3}";
```

Comment:
ohnoes, this hurts. We're gonna convert this project to ES6 so you can use [template strings](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/template_strings)